timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -346,15 +362,17 @@ void NonbondedInteractionGroup<RealType>::execute_device(
 }
 
 template <typename RealType>
-void NonbondedInteractionGroup<RealType>::set_atom_idxs(const std::vector<int> &atom_idxs) {
-    // Allow empty atom indices
-    verify_atom_idxs(N_, atom_idxs, true);
-    std::vector<unsigned int> unsigned_idxs = std::vector<unsigned int>(atom_idxs.begin(), atom_idxs.end());
-
-    std::set<unsigned int> unique_row_atom_idxs(unique_idxs(unsigned_idxs));
-    // compute set of column atoms as set difference
-    std::vector<unsigned int> col_atom_idxs_v = get_indices_difference(N_, unique_row_atom_idxs);
+void NonbondedInteractionGroup<RealType>::set_atom_idxs(
+    const std::vector<int> &row_atom_idxs, const std::vector<int> &col_atom_idxs) {
+    verify_atom_idxs(N_, row_atom_idxs, true);
+    verify_atom_idxs(N_, col_atom_idxs, true);
+
+    std::vector<unsigned int> unsigned_row_idxs = std::vector<unsigned int>(row_atom_idxs.begin(), row_atom_idxs.end());
+    std::set<unsigned int> unique_row_atom_idxs(unique_idxs(unsigned_row_idxs));
```

Comment:
Don't do the same validation here that the constructor does, to make sure they are disjoint.

Also doesn't seem to ensure that the column indices are unique.

Reply:
Updated in https://github.com/proteneer/timemachine/pull/1011/commits/4c1a68a05f88abc2836f5719ba2a28c8db0db076 to use the same code for validation.

Fo local MD, is it correct to say the initializer should have non-zero NR/NC but set_atom_idxs can have 0 NR or NC? Want to make sure to check the right things here.

Reply:
Correct. It is an inconsistency, but for Local MD it is an expectation that it can handle at least N particles, but can also be run on nothing.